apps/worker/src/app/workflow/services/metric-queue.service.ts

```diff
@@ -0,0 +1,125 @@
+const nr = require('newrelic');
+import { WorkerOptions } from 'bullmq';
+import { Inject, Injectable, Logger } from '@nestjs/common';
+import { QueueService } from '@novu/application-generic';
+import { min, max, sum, mean } from 'simple-statistics';
+
+interface IMetric {
+  count: number;
+  total: number;
+  min: number;
+  max: number;
+  sumOfSquares: number;
+}
+
+function sumOfSquares(values: number[]): number {
+  const meanVal = mean(values);
+
+  return sum(values.map((value) => (value - meanVal) ** 2));
+}
+
+@Injectable()
+export class MetricQueueService extends QueueService<Record<string, never>> {
+  constructor(@Inject('BULLMQ_LIST') private token_list: QueueService[]) {
+    super('metric');
+
+    Logger.warn('Metric queue service created');
+
+    this.bullMqService.createWorker(this.name, this.getWorkerProcessor(), this.getWorkerOpts());
+
+    this.bullMqService.worker.on('completed', async (job) => {
+      await this.jobHasCompleted(job);
+    });
+
+    this.bullMqService.worker.on('failed', async (job, error) => {
+      await this.jobHasFailed(job, error);
+    });
+
+    void this.addToQueue('metric-job', {}, '', {
+      repeat: {
+        immediately: true,
```

Comment:
Note that immediatly can cause some weired behaviour when a new image is deployed. Would suggest to keep this on the pattern only.